---
title: "Combining data output for DEBrowser"
output: github_document
---




```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


Loading the necessary packages
```{r Package List}
library(dplyr)
```

```{r Loading in the data}
#Check where I am (using multiple computers, important to be in the correct working directory)
getwd()
setwd("/Users/mnr006/RNASeqProject/Sailfish_output")

#loading in the gene data first
RNA_1_gene <- read.csv("RNAi-1_gene.tsv", header = TRUE, sep = "\t")
RNA_4_gene <- read.csv("RNAi-4_gene.tsv", header = TRUE, sep = "\t")
RNA_3_gene <- read.csv("RNAi-3_gene.tsv", header = TRUE, sep = "\t")

Untreated_1_gene <- read.csv("Un-1_gene.tsv", header = TRUE, sep = "\t")
Untreated_4_gene <- read.csv("Un-4_gene.tsv", header = TRUE, sep = "\t")
Untreated_3_gene <- read.csv("Un-3_gene.tsv", header = TRUE, sep = "\t")


#loading in the trancscript data next
RNA_1_transcript <- read.csv("RNAi-1_transcript.tsv", header = TRUE, sep = "\t")
RNA_4_transcript <- read.csv("RNAi-4_transcript.tsv", header = TRUE, sep = "\t")
RNA_3_transcript <- read.csv("RNAi-3_transcript.tsv", header = TRUE, sep = "\t")

Untreated_1_transcript <- read.csv("Un-1_transcript.tsv", header = TRUE, sep = "\t")
Untreated_4_transcript <- read.csv("Un-4_transcript.tsv", header = TRUE, sep = "\t")
Untreated_3_transcript <- read.csv("Un-3_transcript.tsv", header = TRUE, sep = "\t")


#inspecting structure of each dataset to confirm they look complete/correct/comparable
str(RNA_1_gene)
str(RNA_4_gene)
str(RNA_3_gene)

str(Untreated_1_gene)
str(Untreated_4_gene)
str(Untreated_3_gene)

str(RNA_1_transcript)
str(RNA_4_transcript)
str(RNA_3_transcript)

str(Untreated_1_transcript)
str(Untreated_4_transcript)
str(Untreated_3_transcript)

#confirming that the column names are the same across treatments
colnames(RNA_1_gene)
colnames(Untreated_1_gene)
```


```{r Making the tables}
#This works but lets use dplyr mutate instead
#Basics <- select(RNA_1_gene, c("Name", "NumReads")) %>%
 # bind_cols(RNA_4_gene['TPM'], RNA_3_gene['NumReads'])

#using dplyr to add new columns to gene table
Allsamples_genes <- select(RNA_1_gene, c("Name", "NumReads")) %>%
  mutate(RNAi1 = NumReads, RNAi3 = RNA_3_gene$NumReads, RNAi4 = RNA_4_gene$NumReads, Ctrl1 = Untreated_1_gene$NumReads, Ctrl3 = Untreated_3_gene$NumReads, Ctrl4 = Untreated_4_gene$NumReads) %>%
  select(-NumReads)


#now making transcript table (trying a slightly different way so we don't need to go back and unselect the NumReads from the first part)
Allsamples_transcript <- select(RNA_1_transcript, c("Name", "NumReads")) %>%
  rename(RNAi1=NumReads) %>% 
  mutate(RNAi3 = RNA_3_transcript$NumReads, RNAi4 = RNA_4_transcript$NumReads, Ctrl1 = Untreated_1_transcript$NumReads, Ctrl3 = Untreated_3_transcript$NumReads, Ctrl4 = Untreated_4_transcript$NumReads)

```

```{r Saving tables}
write.table(Allsamples_genes, "Allsamples_genes.tsv", quote = FALSE, sep = "\t", row.names = FALSE)
getwd()
write.table(Allsamples_transcript, "Allsamples_transcript.tsv", quote = FALSE, sep = "\t", row.names = FALSE)
```


```{r Metadata}
#May need to make a new metadata table but it looks like the target.txt files are working appropriately for this 
```




